hr_announcement/models/res_users.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2023 Tecnativa - Pilar Vargas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, models
+
+
+class ResUsers(models.Model):
+    _inherit = "res.users"
+
+    @api.model
+    def announcement_user_count(self):
+        # res = super().announcement_user_count()
+        employee_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "employee"),
+                ("in_date", "=", True),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["employee_ids"],
+        )
+        announcements = self.env["announcement"].browse(
+            [
+                announcement["id"]
+                for announcement in employee_announcements
+                if announcement["employee_ids"]
+                and any(
+                    id in self.env["hr.employee"].search([("user_id", "!=", False)]).ids
+                    for id in announcement["employee_ids"]
+                )
+            ]
+        )
+        department_announcements = self.env["announcement"].search_read(
+            [
+                ("announcement_type", "=", "department"),
+                ("in_date", "=", True),
+                ("id", "not in", self.env.user.read_announcement_ids.ids),
+            ],
+            ["department_ids"],
+        )
+        departments_with_members = self.env["hr.department"].search(
+            [("member_ids.user_id", "!=", False)]
```

Comment:
Why do you search all the departments? You just need to cross the departments in the department_announcements with the departments the user's employee belongs to.